File: bundled/tool/frappe_vscode/models/frappe_app.py
from collections import OrderedDict


class FrappeApp:
    Name: str
    Version: str
    Path: str
    Modules: OrderedDict[str, "FrappeModule"]

    def __init__(self, name, version, path):
        self.Name = name
        self.Version = version
        self.Modules = OrderedDict()
        self.Path = path


class FrappeModule:
    Name: str
    AppName: str
    Path: str
    DocTypes: OrderedDict[str, "FrappeDocType"]
    Reports: OrderedDict[str, "FrappeReport"]

    def __init__(self, name, app_name, path):
        self.Name = name
        self.AppName = app_name
        self.Path = path
        self.DocTypes = OrderedDict()
        self.Reports = OrderedDict()


class FrappeReport:
    Name: str
    ModuleName: str
    Path: str
    IsStandard: bool = False
    ReportType: str
    AppName: str

    def __init__(self, name, report_tpe, module_name, app_name, path):
        self.Name = name
        self.ReportType = report_tpe
        self.ModuleName = module_name
        self.AppName = app_name
        self.Path = path


class FrappeDocType:
    Name: str
    ModuleName: str
    Path: str
    Fields: OrderedDict[str, "FrappeDocTypeField"]
    description: str | None

    def __init__(self, name, module_name, app_name, path):
        self.Name = name
        self.ModuleName = module_name
        self.AppName = app_name
        self.Path = path
        self.Fields = OrderedDict()
        self.description = None


class FrappeDocTypeField:
    Name: str
    Type: str
    Label: str
    Required: bool

    def __init__(self, name, type, label, required):
        self.Name = name
        self.Type = type
        self.Label = label
        self.Required = required

File: src/models/FrappeApp.ts
/* eslint-disable @typescript-eslint/naming-convention */
export class FrappeModule {
    Name: string;
    AppName: string;
    Path: string;
    DocTypes: { [key: string]: FrappeDocType };
    Reports: { [key: string]: FrappeReport };

    constructor(name: string, appName: string, path: string) {
        this.Name = name;
        this.AppName = appName;
        this.Path = path;
        this.DocTypes = {};
        this.Reports = {};
    }
}
export class FrappeReport {
    Name: string;
    ModuleName: string;
    Path: string;

    constructor(name: string, moduleName: string, path: string) {
        this.Name = name;
        this.ModuleName = moduleName;
        this.Path = path;
    }
}
export class FrappeApp {
    Name: string;
    Version: string;
    Path: string;
    Modules: { [key: string]: FrappeModule };

    constructor(name: string, version: string, path: string) {
        this.Name = name;
        this.Version = version;
        this.Path = path;
        this.Modules = {};
    }
}

export class FrappeDocType {
    Name: string;
    ModuleName: string;
    Path: string;
    Fields: { [key: string]: FrappeDocTypeField };

    constructor(name: string, moduleName: string, path: string) {
        this.Name = name;
        this.ModuleName = moduleName;
        this.Path = path;
        this.Fields = {};
    }
}

export class FrappeDocTypeField {
    Name: string;
    Type: string;
    Label: string;
    Required: boolean;

    constructor(name: string, type: string, label: string, required: boolean) {
        this.Name = name;
        this.Type = type;
        this.Label = label;
        this.Required = required;
    }
}

